### PR TITLE
Allow overrides for livenessProbe and readinessProbe actions through helm values

### PR DESCRIPTION
**Pull request by ericwyles**
This pull request moves the 'exec' actions from liveness and readiness probes out to the values.yaml so they can be overridden.

The problem this fixes is when deploying the sonarqube using the helm chart but with hardened container images (for security), the hardened images are often minimal and don't contain extra executables like wget, hostname, etc. 

This allows us to specify an alternate, simpler probe strategy when deploying to one of these hardened environments, but maintains the current behavior as default.

With the current helm chart, we have to maintain a fork of the chart to change the probes for some environments.

Relates to this discussion: https://community.sonarsource.com/t/helm-chart-probes-issue-with-hardened-container-images-no-wget/114377

This change also has the added benefit of consolidating the probe definitions between the StatefulSet and the Deployment types.

------

Please ensure your pull request adheres to the following guidelines:
- [x] explain your motives to contribute this change: what problem you are trying to fix, what improvement you are trying to make
- [ ] Document your Changes in the `CHANGELOG.md` file of the respected chart as well as the `Chart.yaml`

### CODE REVIEW

**Comment by ericwyles** [comment]
Hi @carminevassallo , I saw you assigned this to yourself. Please let me know if you have any questions or suggestions. Thank you!

**Comment by carminevassallo** [comment]
Hi @ericwyles,

yes, I will review it and provide you with my feedback by the end of this week :)

File: charts/sonarqube/templates/deployment.yaml [hunk 1]
```diff
@@ -270,29 +270,9 @@ spec:
...
-              - |
-                #!/bin/bash
-                # A Sonarqube container is considered ready if the status is UP, DB_MIGRATION_NEEDED or DB_MIGRATION_RUNNING
-                # status about migration are added to prevent the node to be kill while sonarqube is upgrading the database.
-                if wget --no-proxy -qO- http://localhost:{{ .Values.service.internalPort }}{{ .Values.readinessProbe.sonarWebContext | default (include "sonarqube.webcontext" .) }}api/system/status | grep -q -e '"status":"UP"' -e '"status":"DB_MIGRATION_NEEDED"' -e '"status":"DB_MIGRATION_RUNNING"'; then
-                	exit 0
-                fi
-                exit 1
+{{ tpl (toYaml .Values.readinessProbe | indent 12) . }}
             initialDelaySeconds: {{ .Values.readinessProbe.initialDelaySeconds }}
```
Comment (carminevassallo) [inline charts/sonarqube/templates/deployment.yaml:276]: Please make sure the following `readinessProbe`  thresholds

**Review by carminevassallo — CHANGES_REQUESTED** [review]
Hey @ericwyles,

Thanks for your patience!

I was triggering the CI on your branch, and there seem to be issues during the static compatibility tests. Can you please check those tests and possibly adjust the outdated values?

Apart from that, I was reviewing your PR locally and didn't find any issues with it (apart from a small typo in the deployment template). It's actually a great contribution :)

Could you please add an entry in the `CHANGELOG.md` and `Chart.yaml` (`artifacthub.io/changes`) describing your contribution?

File: charts/sonarqube/templates/deployment.yaml (hunk 1 above)
Comment (ericwyles) [reply charts/sonarqube/templates/deployment.yaml:276]: Thanks @carminevassallo I'll get this fixed up!

Comment (ericwyles) [reply charts/sonarqube/templates/deployment.yaml:276]: @carminevassallo It looks like the CHANGELOG.md entries are organized by version, should I put my updates in the 10.6.0 section?

Comment (ericwyles) [reply charts/sonarqube/templates/deployment.yaml]: @carminevassallo I figured it out. This should be ready for another look whenever you get a chance. Thanks!

**Comment by ericwyles** [comment]
Hi @carminevassallo , I found an additional error and updated the templating so that sonarWebContext is still available to be used in the probes but doesn't get rendered in the templated yaml. (Used 'omit' on it). That should clear up the last syntax error I was able to find with the probes running the unit_helm_compatibility_test.sh locally.

In addition, I added a unit compatability test for 'deploymentType: Deployment' in an additional test file. I did this because I had the same syntax error in StatefulSet as well as Deployment, but only the StatefulSet one was being caught. I think this is ready for a re-review now.

**Comment by ericwyles** [comment]
Hi @carminevassallo , checking back on this to see if we can get it re-reviewed and hopefully merged. Thanks again!

**Review by carminevassallo — APPROVED** [review]
Hey @ericwyles,

Thanks for your patience. As you might know, we are focused on our own roadmap and review external contributions on a best effort basis.

The contribution is ready to be merged, we are gonna do something similar for the DCE edition. I tested the whole templating locally as well and your proposed changes work as intended :) Thanks also for the last minute fix of the typo, I was about to push the fix but you were already on it xD That's super cool!

I'm gonna squash and merge...

**Comment by ericwyles** [comment]
Thanks @carminevassallo ! I totally understand and was just doing a periodic check in. Thanks for all your help and patience and getting this merged. I'll be on the lookout for the 10.6.0 release so we can adopt this in our project. Thanks again!

**Pull request merged**